src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -125,10 +133,10 @@ def ruby_response(response)
   end
 
   def create_client_if_nil
-    if @client.nil?
+    if @@client.nil?
       client_options = ClientOptions.new
       configure_ssl(client_options)
-      @client = SyncHttpClient.createClient(client_options)
+      @@client = SyncHttpClient.createClient(client_options)
```

Comment:
This whole method should become a class method; i.e. `def self.create_client`.  Then you don't need the double-@ here.